#VRML_SIM R2025a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/appearances/protos/Grass.proto"
EXTERNPROTO "../protos/E-puck.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/appearances/protos/Asphalt.proto"

WorldInfo {
}
Viewpoint {
  orientation 0.3464518914795183 0.08396724592685598 -0.9343021933516602 2.787855574212413
  position 2.692239472715303 1.155547514135325 2.205775239290684
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 10 10
  floorAppearance Grass {
    type "prickly"
  }
}
Pose {
  translation 1.05859 0.0714216 0
  rotation 0 0 1 0.785398
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0 0
        }
      }
      geometry Box {
        size 0.2 0.05 0.0001
      }
      castShadows FALSE
      isPickable FALSE
    }
  ]
}
Pose {
  translation -1.05568 0.502765 -3.8359e-10
  rotation -9.58979e-09 2.31518e-08 -1 -0.785395307179586
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0 0
        }
      }
      geometry Box {
        size 0.2 0.05 0.0001
      }
      castShadows FALSE
      isPickable FALSE
    }
  ]
}
Pose {
  translation 1.06566 0.502756 0
  rotation 0 0 1 -0.785395307179586
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0 0
        }
      }
      geometry Box {
        size 0.2 0.05 0.0001
      }
      castShadows FALSE
      isPickable FALSE
    }
  ]
}
Pose {
  translation -1.06275 0.0714129 0
  rotation 0 0 1 -0.785395307179586
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0 0
        }
      }
      geometry Box {
        size 0.2 0.05 0.0001
      }
      castShadows FALSE
      isPickable FALSE
    }
  ]
}
Pose {
  translation 1.1327 0.288186 0
  rotation 0 0 -1 -1.5708053071795867
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0 0
        }
      }
      geometry Box {
        size 0.3 0.05 0.0001
      }
      castShadows FALSE
      isPickable FALSE
    }
  ]
}
Pose {
  translation -1.1273 0.288169 0
  rotation 0 0 -1 -1.5708053071795867
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0 0
        }
      }
      geometry Box {
        size 0.3 0.05 0.0001
      }
      castShadows FALSE
      isPickable FALSE
    }
  ]
}
Pose {
  translation 0 0.57 0
  rotation 1 0 0 0
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0 0
        }
      }
      geometry Box {
        size 2 0.05 0.0001
      }
      castShadows FALSE
      isPickable FALSE
    }
  ]
}
Pose {
  rotation 1 0 0 0
  children [
    Shape {
      appearance Asphalt {
      }
      geometry Box {
        size 2 0.05 0.0001
      }
      castShadows FALSE
      isPickable FALSE
    }
  ]
}
DEF Nice_Epuck E-puck {
  translation 0.08519994538743557 1.0979237955826167e-07 -6.497542027386095e-05
  rotation 2.200228898912202e-05 -0.9999981949463199 -0.0018998999977157018 0.00011026584975874844
  name "Nice_Epuck"
  controller "gp_controller"
  emitter_channel 4
  receiver_channel 3
  groundSensorsSlot [
    DistanceSensor {
      translation 0.003 0.015 0.01
      rotation 0 -1 0 -1.5708
      name "gs0"
      lookupTable [
        0 4095 0
        0.04 1000 0
        0.05 800 0
        0.06 500 0
        0.08 200 0
        0.1 0 0
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation 0.003 0 0.01
      rotation 0 -1 0 -1.5708
      name "gs1"
      lookupTable [
        0 4095 0
        0.04 1000 0
        0.05 800 0
        0.06 500 0
        0.08 200 0
        0.1 0 0
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation 0.003 -0.015 0.01
      rotation 0 -1 0 -1.5708
      name "gs2"
      lookupTable [
        0 4095 0
        0.04 1000 0
        0.05 800 0
        0.06 500 0
        0.08 200 0
        0.1 0 0
      ]
      type "infra-red"
    }
  ]
}
DEF Supervisor Robot {
  translation 0 0.3 0
  children [
    Receiver {
      channel 4
    }
    Emitter {
      channel 3
    }
  ]
  name "Supervisor"
  controller "ga_supervisor"
  supervisor TRUE
}
